### ISSUE TITLE
Improve user notifications system

### ISSUE THREAD

**Issue by paglias** [HabitRPG/habitica#7617]
#7588 introduced user notifications stored in `user.notifications` in order to be able to remove `localStorage`. We only implemented the notifications that previously relies on the presence of `localStorage`. In future we'd like greatly to expand the use of these notifications.

Some ideas / things to discuss:
- [ ] move drops to notifications from user._tmp
- [ ] introduce additional properties to each notification (for example a localized message, some additional  data)
- [ ] introduce ephemeral notifications that are only sent once but not stored in the database for changes in user's stats
- [ ] implement long polling / realtime
- [ ] use them to notify the user of invitations to groups, new messages, .... but only as notifications, marking its notification as read should not reject a party invitation for example

**Comment by crookedneighbor** [comment]
What happens when a 3rd party scores a task and a drop gets added to the queue? Would that mean that the next time I use the website, I'd be inundated with drop notifications? Or would this be part of the ephemeral notifications type?

**Comment by paglias** [comment]
We could batch them... so if you happen to login and have 10 drops, you'd only get 1 notification

**Comment by librarianmage** [comment]
Should this still be on hold?

**Comment by paglias** [comment]
Yes let's keep this on hold as a bigger discussion needs to happen before

**Comment by Alys** [comment]
@TheHollidayInn is looking into this along with related problems.

**Comment by paglias** [comment]
Most of this issues will be fixed in #9716
